gems/pending/util/miq-password.rb

```diff
@@ -143,22 +150,16 @@ def self.v2_key
   end
 
   def self.legacy_keys
-    [v1_key, v0_key].compact
+    @@legacy_keys ||= []
   end
 
   def self.add_legacy_key(filename, type = :v1)
-    case type
-    when :v0
-      @v0_key = ez_load(filename, false)
-    when :v1
-      @v1_key = ez_load(filename)
-    end
+    key = ez_load(filename, type != :v0)
+    legacy_keys << key if key && !legacy_keys.include?(key)
+    key
   end
 
   class << self
-    attr_accessor :v0_key
-    attr_accessor :v1_key
-
     def v2_key=(key)
```

Comment:
I looked at what it would take to do this, and it is minor.

it sure would be bandaid style.
I'd like input from others before I do this